plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,502 @@
+    - The UUID of the static report route map.
+    type: str
+    aliases: [ static_report_route_map_for_multicast ]
+  static_report_route_map:
+    description:
+    - The Route Map Policy for Multicast.
+    - This parameter can be used instead of O(static_report_route_map_uuid).
+    - If both parameter are used, O(static_report_route_map) will be ignored.
+    type: dict
+    suboptions:
+      name:
+        description:
+        - The name of the Route Map Policy for Multicast.
+        type: str
+        required: true
+      template:
+        description:
+        - The name of the template in which the Route Map Policy for Multicast has been created.
+        type: str
+        required: true
+    aliases: [ static_report_route_map_policy, static_report_route_map_policy_multicast ]
+  maximum_multicast_entries:
+    description:
+    - The maximum multicast entries value.
+    - Defaults to C(4294967295) when unset during creation.
+    - The valid range is from C(1) to C(4294967295).
+    - This parameter is only applicable when the O(state_limit_route_map_uuid) is not empty.
+    type: int
+  reserved_multicast_entries:
+    description:
+    - The reserved multicast entries value.
+    - Defaults to C(0) when unset during creation.
+    - The valid range is from C(0) to C(4294967295).
+    type: int
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the Tenant template.
+- The O(state_limit_route_map_uuid), O(report_policy_route_map_uuid), O(static_report_route_map_uuid) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_tenant_route_map_policy_for_multicast) to create the Route Map Policy for Multicast.
+seealso:
+- module: cisco.mso.ndo_template
+- module: cisco.mso.ndo_tenant_route_map_policy_for_multicast
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create an IGMP Interface Policy
+  cisco.mso.ndo_tenant_igmp_interface_policy:
+    template: TestTenantTemplate
+    name: test_igmp_interface_policy
+    description: Test Description
+    version3_asm: enabled
+    fast_leave: enabled
+    report_link_local_groups: enabled
+    igmp_version: v3
+    group_timeout: 260
+    query_interval: 125
+    query_response_interval: 10
+    last_member_count: 2
+    last_member_response_time: 1
+    startup_query_count: 2
+    startup_query_interval: 31
+    querier_timeout: 255
+    robustness_variable: 2
+    state_limit_route_map_uuid: route_map_policy_for_multicast_uuid
+    report_policy_route_map_uuid: route_map_policy_for_multicast_uuid
+    static_report_route_map:
+      name: TestStaticReportRouteMap
+      template: TestTenantTemplate
```

Comment:
Is this used anywhere- **template: TestTenantTemplate**?

Reply:
changed to ansible_tenant_template